src/storage/db_interface_frontend.py

```diff
@@ -106,10 +110,18 @@ def _get_one_virtual_path_of_fo(fo_dict, root_uid):
         return FileObject.get_top_of_virtual_path(fo_dict['virtual_file_path'][root_uid][0])
 
     def _get_hid_firmware(self, uid):
-        firmware = self.firmwares.find_one({'_id': uid}, {'vendor': 1, 'device_name': 1, 'version': 1, 'device_class': 1})
+        firmware = self.firmwares.find_one({'_id': uid}, {'vendor': 1, 'device_name': 1, 'device_part': 1, 'version': 1, 'device_class': 1})
         if firmware is not None:
-            return '{} {} - {} ({})'.format(firmware['vendor'], firmware['device_name'], firmware['version'], firmware['device_class'])
-        return None
+            if 'device_part' not in firmware:  # for compatibility reasons
+                part = ' -'
+            else:
+                if firmware['device_part'] != '':
```

Comment:
`part = ' - {}'.format(firmware['device_part']) if firmware['device_part'] != '' else ' -'`

or even

`part = ' -' if 'device_part' not in firmware or firmware['device_part'] == '' else ' - {}'.format(firmware['device_part'])`